draft-ietf-mls-protocol.md

```diff
@@ -1804,29 +1810,33 @@ struct {
 } ParentHashInput;
 ~~~~~
 
-The Parent Hash of P with Co-Path Child S is the hash of a `ParentHashInput` object
-populated as follows. The field `public_key` contains the HPKE public key of P. If P
-is the root, then `parent_hash` is set to a zero-length octet string.
-Otherwise `parent_hash` is the Parent Hash of P's parent with P's sibling as the
-co-path child.
+The field `public_key` contains the HPKE public key of P. If P is the root,
+then the `parent_hash` field is set to a zero-length octet string. Otherwise,
+`parent_hash` is the Parent Hash of the next node after P on the filtered
+direct path of U (see {{resolution-example}}). This way, P's Parent Hash fixes
+all new HPKE public keys of nodes on the path from P to the root.
 
-Finally, `original_child_resolution` is the array of `HPKEPublicKey` values of the
+Finally, `original_child_resolution` is the array of HPKE public keys of the
 nodes in the resolution of S but with the `unmerged_leaves` of P omitted. For
 example, in the ratchet tree depicted in {{resolution-example}} the
 `ParentHashInput` of node Z with co-path child C would contain an empty
 `original_child_resolution` since C's resolution includes only itself but C is also
 an unmerged leaf of Z. Meanwhile, the `ParentHashInput` of node Z with co-path child
 D has an array with one element in it: the HPKE public key of D.
+This way, P's Parent Hash fixes the set of HPKE public keys to which the new HPKE
```

Comment:
A pretty similar sentence is already in the paragraph above.

Reply:
It's similar, but there's a key difference: the paragraph above considers attesting to the public key V.pk for each node V on the path. This paragraph considers attesting to the public keys U.pk to which V.sk was encrypted.

Maybe giving names to nodes would make it clearer? Like "This way, P's parent hash fixes V.pk for each V on the filtered path from P to the root" and then "This way, P's parent hash fixes all public key to which V.pk was encrypted for each V[...]"?